cmd/soroban-cli/src/config/secret.rs

```diff
@@ -42,15 +42,31 @@ pub struct Args {
     /// (deprecated) Enter key using 12-24 word seed phrase
     #[arg(long)]
     pub seed_phrase: bool,
+
+    /// Add using a key saved in a secure store entry. Requires the entry name to be provided with `--entry_name`
+    #[arg(long)]
+    pub secure_store: bool,
 }
 
 impl Args {
-    pub fn read_secret(&self) -> Result<Secret, Error> {
+    pub fn read_secret(&self, name: &KeyName) -> Result<Secret, Error> {
         if let Ok(secret_key) = std::env::var("SOROBAN_SECRET_KEY") {
-            Ok(Secret::SecretKey { secret_key })
+            return Ok(Secret::SecretKey { secret_key });
+        };
+
+        println!("Type a secret key or 12/24 word seed phrase:");
+        let secret_key = read_password()?;
+
+        if self.secure_store {
+            // TODO: if secret_key is a private key, this will return error for now
```

Comment:
Curious to hear what others think - but I think that we'll be able to let user's create new keys that are stored in the OS secure storage, and the values saved can be either private keys or seed phrases. That is not quite supported yet here, but I think its possible. Does that make sense, or is it too confusing?

The only downside i'm seeing right now is that when we add the ability to fetch the seed phrase for a key, we'll need to return an error if the user entered their private key directly.